website/client/src/components/groups/questDetailModal.vue

```diff
@@ -0,0 +1,450 @@
+    QuestRewards,
+    questDialogContent,
+    closeIcon,
+    SelectTranslatedArray,
+    QuestInfo,
+    Item,
+  },
+  mixins: [questActionsMixin],
+  props: ['group'],
+  data () {
+    return {
+      loading: false,
+      selectMode: true,
+      selectedQuest: {},
+      fromSelectionDialog: false,
+      icons: Object.freeze({
+        navigationBack,
+      }),
+      shareUserIdShown: false,
+      quests,
+      sortBy: 'AZ',
+    };
+  },
+  computed: {
+    ...mapState({ user: 'user.data' }),
+    questData () {
+      return quests.quests[this.selectedQuest];
+    },
+    dialogClass () {
+      if (!this.groupHasQuest || this.fromSelectionDialog || this.canEditQuest) {
+        return 'need-bottom-padding';
+      }
+
+      return '';
+    },
+
+    questsInfoList () {
+      const availableQuests = Object.entries(this.user.items.quests)
+        .filter(([, amount]) => amount > 0);
+
+      return orderBy(availableQuests.map(([key, amount]) => {
+        const questItem = quests.quests[key];
+
+        return {
+          ...getItemInfo(this.user, 'quests', questItem),
+          amount,
+        };
+      }), item => (this.sortBy === 'AZ' ? item.text : -item.amount));
+    },
+  },
+  mounted () {
+    const userQuests = this.user.items.quests;
+    for (const key in userQuests) {
+      if (userQuests[key] > 0) {
+        this.selectedQuest = key;
+        break;
+      }
+    }
+
+    this.$root.$on('bv::show::modal', this.handleOpen);
+  },
+  beforeDestroy () {
+  },
+  methods: {
+    selectQuest (selectQuestPayload) {
+      this.selectMode = false;
+      this.selectedQuest = selectQuestPayload.key;
+      this.fromSelectionDialog = true;
+    },
+    async questInit () {
+      this.loading = true;
+
+      Analytics.updateUser({
+        partyID: this.group._id,
+        partySize: this.group.memberCount,
+      });
+
+      const groupId = this.group._id || this.user.party._id;
+
+      const key = this.selectedQuest;
+      try {
+        const response = await this.$store.dispatch('guilds:inviteToQuest', { groupId, key });
+        const quest = response.data.data;
+
+        // TODO move the state updates to the action itself
+        const partyState = this.$store.state.party;
+
+        if (!partyState.data) {
+          partyState.data = {};
+        }
+
+        partyState.data.quest = quest;
+      } finally {
+        this.loading = false;
+      }
+
+      this.close();
+    },
+    close () {
+      this.$root.$emit('bv::hide::modal', 'quest-detail-modal');
+    },
+    async questConfirm () {
+      const accepted = await this.questActionsConfirmQuest();
+
+      if (accepted) {
+        this.close();
+      }
+    },
+    async questCancel () {
+      const accepted = await this.questActionsCancelOrAbortQuest();
+
+      if (accepted) {
+        this.close();
+      }
+    },
+    goBackToQuestSelection () {
+      this.selectMode = true;
+    },
+    handleOpen (_, selectQuestPayload) {
```

Comment:
The underscore as a placeholder parameter is a weird pattern, what's that for?

Reply:
since the first param isn't used - it can't stay there as unused

_  is usually the pattern for "don't mind me, I'm unused" .... well except for lodash but thats a different story :D